app/src/main/java/com/duckduckgo/app/privatesearch/PrivateSearchViewModel.kt

```diff
@@ -49,16 +50,22 @@ class PrivateSearchViewModel @Inject constructor(
 ) : ViewModel() {
 
     data class ViewState(
-        val autoCompleteSuggestionsEnabled: Boolean = true,
-        val autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled: Boolean = true,
-        val storeHistoryEnabled: Boolean = false,
+        val autoCompleteSuggestionsEnabled: Boolean,
+        val autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled: Boolean,
+        val storeHistoryEnabled: Boolean,
     )
 
     sealed class Command {
         object LaunchCustomizeSearchWebPage : Command()
     }
 
-    private val viewState = MutableStateFlow(ViewState())
+    private val viewState = MutableStateFlow(
```

Comment:
any reason why we are doing this here?
Now on class init, we are querying the different data sources and we can block main thread (see runblocking), producing ANR.

Reply:
Yes, I’m doing this on purpose. If there are default values to start with, it causes two issues:
- It messes up the activity transition
- You see the switch changing state upon entering the screen

The `runBlocking` is on a method that gets a shared pref boolean (Which shouldn’t be a `suspend` function IMO), so it won’t cause ANRs.

The only other option is to show some sort of loading indicator (Which is not a pattern we follow) or hide the screen until it completes (Which won't look good).

I have recordings here of the transition/toggle behavior: https://app.asana.com/0/0/1207888184745508/1207887624571935/f

Reply:
I’ve moved the initialization of `viewState` into `init` using a coroutine (https://github.com/duckduckgo/Android/pull/4783/commits/0fc4ea802b1e9e3e8b7b733c76322beedd6da3c3)

Reply:
Some notes from our zoom call on Friday:

- If we need to access database or shared prefs to initialize a value as part of creating a class instance, that can produce an ANR since it happens on main thread.
- Not having an "idle" or "empty" state in UI forces us to define a default value in the code. That default value can be defined in the viewmodel (e.g: default viewstate value) or in the layout (e.g: default state of the view we put, including a toggle).
- `init` is just the early point we can execute a background query without blocking main thread, so we can try to query the datasource and get a value before layout is rendered. Reducing the chances of seeing the toggle changing state in the transition.